### ISSUE TITLE
Support Request: WSL2 mkcert config

### ISSUE THREAD

**Issue by MrPaulDriver** [drud/ddev#2397]
Following along with the blog post at https://www.ddev.com/ddev-local/ddev-wsl2-getting-started/

Running into a problem with the CAROOT environment variable commands.

`setx CAROOT "$(mkcert -CAROOT)"; If ($Env:WSLENV -notlike "*CAROOT*") { setx WSLENV "CAROOT/up;$Env:WSLENV" }`

returns the following;

`-bash: syntax error near unexpected token `$Env:WSLENV'`

Is the setx CAROOT code snippet incorrect or am I doing something wrong?

I'm new to WSL2 and it's a long time since being a windows user :-)

**Comment by rfay** [comment]
I apologize - we're working on this over in https://github.com/drud/ddev/pull/2375 but didn't get it finished this week due to Drupalcon. Also trying to wrap it all into a PowerShell script at https://github.com/drud/ddev/pull/2390

Your particular problem right now is that you have to run that in PowerShell, not bash. However, if WSLENV is already set, it's a colon-separated string (not comma-separated as was shown in the blog (I just updated it with a colon), so should be 

`setx CAROOT "$(mkcert -CAROOT)"; If ($Env:WSLENV -notlike "*CAROOT*") { setx WSLENV "CAROOT/up:$Env:WSLENV" }`

**Comment by MrPaulDriver** [comment]
Thank you Randy, that appears to have resolved the issue.

The documentation for installing mkcert leads with using a PowerShell window so a change to the docs will save others getting stuck like I did.

Thank you once again for great support.

**Comment by rfay** [comment]
Thanks - would appreciate if you could scan the latest version of the docs PR (https://github.com/drud/ddev/blob/8bb15aad74ed6b2495373448d0fc2b25b32e2e0c/docs/index.md) (that's from https://github.com/drud/ddev/pull/2375) 

Also if you could take a quick look at the blog post and see if it is looking right to you. https://ddev.com/ddev-local/ddev-wsl2-getting-started/

**Comment by MrPaulDriver** [comment]
Yes, I'll be very happy to do that.

As you may seen from this [stack exchange post](https://stackoverflow.com/questions/62992034/why-does-composer-create-project-for-drupal-fail-in-wsl2/), the very last sentence of the revised WSL2 docs turned out to be a vital part of the process and one that I had not figured out for myself.

> Make sure you put your projects in the Linux filesystem (e.g. /home/<your_username>), not in the Windows filesystem (/mnt/c), because you'll get vastly better performance on the Linux filesystem.

**Comment by MrPaulDriver** [comment]
I reviewed the docs and they look great

**Comment by rfay** [comment]
Thanks so much!

**Comment by MrPaulDriver** [comment]
For those like myself who are not yet up to speed, you might want to include a caution about being careful about using Windows tools to open folders and files in the Linux file system. [Do not change Linux files using Windows apps and tools.](https://devblogs.microsoft.com/commandline/do-not-change-linux-files-using-windows-apps-and-tools/)

I'm now installing [Windows 10 Insider Preview Build 19603](https://blogs.windows.com/windowsexperience/2020/04/08/announcing-windows-10-insider-preview-build-19603/), available in the Dev channel. Apparently this does include a Linux compatible file explorer. 

I am thinking that a visual file browser will be pretty essential for inspecting projects, plus dragging folders and files into code editors etc . I expect most Windows users would expect the same.

**Comment by rfay** [comment]
The regular Explorer is fine for graphical exploration. As your [link](https://devblogs.microsoft.com/commandline/do-not-change-linux-files-using-windows-apps-and-tools/) says at the top, 

> Update – May 2019 Windows 10 version 1903 has now been released and allows an important update to WSL that allows Windows applications and tools to access Linux files directly. To do this, WSL hosts a new 9P fileserver, which exposes distro filesystems to Windows apps and tools via \\wsl$\<DistroName>\!

You still aren't supposed to use %LOCALAPPDATA%, but that's basically obsolete WSL1 usage. You can use \\wsl$ just fine.  You can also use it in PHPStorm, works fine, but slow building cache. 

You may be interested in [DDEV-Local and PHPStorm](https://ddev.com/ddev-local/ddev-local-and-phpstorm-debugging-with-wsl2/) article.

![Annotation 2020-07-20 062531](https://user-images.githubusercontent.com/112444/87937428-f3d1dc00-ca51-11ea-8d94-3a0b07518c0e.png)

**Comment by MrPaulDriver** [comment]
As mentioned above I was keen to try the Windows dev channel, but found it a bit too unstable for my liking, so I reinstalled Windows and started from scratch using the notes from https://github.com/drud/ddev/blob/8bb15aad74ed6b2495373448d0fc2b25b32e2e0c/docs/index.md

Everything seemed okay until I ran ddev launch, at which point a message was shown suggesting that mkcert was not correctly installed. I uninstalled mkcert and reinstalled, but found the same problem again.

From @dennisameling's original post at https://gist.github.com/dennisameling/8317b9dc6b7d971860a4797c64f16eaf#prepare-ssl-certificate-mkcert I saw the steps below were different to those mentioned in the docs.

> Run export CAROOT=/mnt/c/Users/YOUR_WINDOWS_USERNAME/AppData/Local/mkcert. This will make sure that mkcert on Linux uses your Windows certificates when issuing SSL certificates, so you don't get SSL warnings in your browser on Windows.
Run echo 'export CAROOT=/mnt/c/Users/YOUR_WINDOWS_USERNAME/AppData/Local/mkcert' >> ~/.profile to store this in your profile for future sessions.

After applying these steps I found that mkcert now works correctly.

I am guessing the PowerShell script is intended to streamline the export steps, but I have some doubts this is working correctly, based on my experience of a fresh Windows installation and following the docs to the letter. 

Now I have it working, I am a bit reluctant to back track for testing, but hope it is okay to reopen the issue.
